### PR TITLE
#82 error response api

### PR DESCRIPTION
**Pull request by rahulraghu94**
Using the Error Response API in place of manually checking for error codes in examples

### CODE REVIEW

**Comment by rahulraghu94** [comment]
@dhoomakethu , all tests using the error API

File: examples/contrib/remote_server_context.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def validate(self, fx, address, count=1):
         result = self.context.get_callbacks[self.decode(fx)](address,
                                                              count,
                                                              self.unit_id)
-        return result.function_code < 0x80
+        return result.isError()
```
Comment (dhoomakethu) [inline examples/contrib/remote_server_context.py]: This should be `return not result.isError()`

File: examples/functional/base_runner.py [hunk 2]
```diff
@@ -30,33 +30,33 @@ def shutdown(self):
     def testReadWriteCoil(self):
         rq = self.client.write_coil(1, True)
         rr = self.client.read_coils(1,1)
-        self.__validate(rq, lambda r: r.function_code < 0x80)
+        self.__validate(rq, lambda r: r.isError())
```
Comment (dhoomakethu) [inline examples/functional/base_runner.py]: The check should be for `not isError()`

File: examples/functional/base_runner.py [hunk 3]
```diff
@@ -30,33 +30,33 @@ def shutdown(self):
...
         self.__validate(rr, lambda r: r.bits[0] == True)
-        
+
     def testReadWriteCoils(self):
         rq = self.client.write_coils(1, [True]*8)
         rr = self.client.read_coils(1,8)
-        self.__validate(rq, lambda r: r.function_code < 0x80)
+        self.__validate(rq, lambda r: r.isError())
```
Comment (dhoomakethu) [inline examples/functional/base_runner.py]: same as above

File: examples/functional/base_runner.py [hunk 4]
```diff
@@ -30,33 +30,33 @@ def shutdown(self):
...
         self.__validate(rr, lambda r: r.bits == [True]*8)
-        
+
     def testReadWriteDiscreteRegisters(self):
         rq = self.client.write_coils(1, [False]*8)
         rr = self.client.read_discrete_inputs(1,8)
-        self.__validate(rq, lambda r: r.function_code < 0x80)
+        self.__validate(rq, lambda r: r.isError())
```
Comment (dhoomakethu) [inline examples/functional/base_runner.py]: same as above

File: examples/functional/base_runner.py [hunk 5]
```diff
@@ -30,33 +30,33 @@ def shutdown(self):
...
         self.__validate(rr, lambda r: r.bits == [False]*8)
-        
+
     def testReadWriteHoldingRegisters(self):
         rq = self.client.write_register(1, 10)
         rr = self.client.read_holding_registers(1,1)
-        self.__validate(rq, lambda r: r.function_code < 0x80)
+        self.__validate(rq, lambda r: r.isError())
```
Comment (dhoomakethu) [inline examples/functional/base_runner.py]: same as above

File: pymodbus/datastore/remote.py [hunk 6]
```diff
@@ -41,7 +41,7 @@ def validate(self, fx, address, count=1):
         '''
         _logger.debug("validate[%d] %d:%d" % (fx, address, count))
         result = self.__get_callbacks[self.decode(fx)](address, count)
-        return result.function_code < 0x80
+        return True if result.isError() is False else False
```
Comment (dhoomakethu) [inline pymodbus/datastore/remote.py]: simplify `return not result.isError()`

**Comment by rahulraghu94** [comment]
@dhoomakethu , this is ready now.

**Pull request merged**